metadata-io/src/main/java/com/linkedin/metadata/client/JavaEntityClient.java

```diff
@@ -570,4 +578,34 @@ private void tryIndexRunId(Urn entityUrn, @Nullable SystemMetadata systemMetadat
             _entitySearchService.appendRunId(entityUrn.getEntityType(), entityUrn, systemMetadata.getRunId());
         }
     }
+
+    protected <T> T withRetry(@Nonnull final Supplier<T> block) {
+        final BackoffPolicy backoffPolicy = new ExponentialBackoff(DEFAULT_RETRY_INTERVAL);
+        int attemptCount = 0;
+
+        while (attemptCount < DEFAULT_RETRY_COUNT + 1) {
+            try {
+                return block.get();
+            } catch (Throwable ex) {
+                MetricUtils.counter(JavaEntityClient.class, "exception" + MetricUtils.DELIMITER + ex.getClass().getName().toLowerCase()).inc();
```

Comment:
Thoughts on this counter?

Reply:
ooo love the counter. it might be nice to know what exactly is failing instead of just the name of the exception class.

what do you think about passing a method name or something to track with this counter into `withRetry`? so here it could be like `delete_references_to_failures` or something - and it could be optional so if they don't provide a metric name we could just pass in the exception class like you're doing